cmd/ndm_daemonset/probe/uuid.go

```diff
@@ -84,6 +85,15 @@ func generateUUID(bd blockdevice.BlockDevice) (string, bool) {
 		klog.Infof("device(%s) has a filesystem, using filesystem UUID: %s", bd.DevPath, bd.FSInfo.FileSystemUUID)
 		uuidField = bd.FSInfo.FileSystemUUID
 		ok = true
+	case features.FeatureGates.IsEnabled(features.PartitionTableUUID) && len(bd.PartitionInfo.PartitionTableType) > 0:
+		if len(bd.PartitionInfo.PartitionTableUUID) == 0 {
+			klog.Errorf("device(%s) has a partition table, but can not get partition table uuid", bd.DevPath)
+			break
```

Comment:
What will happen in the following case:

1. The `break` gets executed here, which means the uuid wont be generated. 
2. NDM goes onto create a partition on the disk, but it will fail since a partition table already exists
3. This will trigger a rescan and causes an infinite loop

Can we have some checks for the above case also, possibly in the eventhandler or addhandler?

Reply:
The GPT partition table without UUID is very weird. But This may be happened. Or can we ignore this disk ?